On macos 10.15:

```
builder for '/nix/store/3l5wjhd8rygd565hlgl5bwaympkwcgdy-python3.9-pygeos-0.12.0.drv' failed with exit code 1; last 10 log lines:
  -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
  =========================== short test summary info ============================
  FAILED pygeos/tests/test_constructive.py::test_float_arg_array[offset_curve-geometry0]
  FAILED pygeos/tests/test_constructive.py::test_float_arg_array[offset_curve-geometry3]
  FAILED pygeos/tests/test_constructive.py::test_float_arg_array[offset_curve-geometry4]
  FAILED pygeos/tests/test_constructive.py::test_float_arg_array[offset_curve-geometry5]
  FAILED pygeos/tests/test_constructive.py::test_float_arg_array[offset_curve-geometry6]
  FAILED pygeos/tests/test_constructive.py::test_float_arg_array[offset_curve-geometry7]
  FAILED pygeos/tests/test_constructive.py::test_float_arg_array[offset_curve-geometry8]
  = 7 failed, 2614 passed, 22 skipped, 1 xfailed, 22 xpassed, 5 warnings in 5.63s =
```

```
builder for '/nix/store/7kg4maa70cahiyw8nr9njhd2jvr3rcr1-python3.9-Shapely-1.8.2.drv' failed with exit code 1; last 10 log lines:
      raise GeometryTypeError("Splitting a Polygon with a %s is not supported" % splitter.type)
  
  tests/test_substring.py::SubstringTestCase::test_raise_type_error
    /nix/store/m1lyy5x531vyi8sxiy3zjabddva72f5q-python3.9-Shapely-1.8.2/lib/python3.9/site-packages/shapely/ops.py:639: ShapelyDeprecationWarning: GeometryTypeError will derive from ShapelyError and not TypeError or ValueError in Shapely 2.0.
      raise GeometryTypeError("Can only calculate a substring of LineString geometries. A %s was provided." % geom.type)
  
  -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
  =========================== short test summary info ============================
  FAILED tests/test_parallel_offset.py::OperationsTestCase::test_parallel_offset_linestring
  ===== 1 failed, 463 passed, 18 deselected, 2 xfailed, 27 warnings in 1.97s =====
```

Same for the python 3.10 versions of those packages

--- a/pkgs/development/python-modules/pygeos/default.nix
+++ b/pkgs/development/python-modules/pygeos/default.nix
@@ -17,6 +17,12 @@ buildPythonPackage rec {
     sha256 = "sha256-PEFULvZ8ZgFfRDrj5uaDUDqKIh+cJPsjgPauQq7RYAo=";
   };
 
+  patches = [
+    # Adapt https://github.com/shapely/shapely/commit/4889bd2d72ff500e51ba70d5b954241878349562,
+    # backporting to pygeos
+    ./fix-for-geos-3-11.patch
+  ];
+
   nativeBuildInputs = [
     geos # for geos-config
     cython

--- a/pkgs/development/python-modules/pygeos/fix-for-geos-3-11.patch
+++ b/pkgs/development/python-modules/pygeos/fix-for-geos-3-11.patch
@@ -1,0 +1,29 @@
+From ea82dbefeb573150935eb78a6916813775512e76 Mon Sep 17 00:00:00 2001
+From: Joris Van den Bossche <jorisvandenbossche@gmail.com>
+Date: Tue, 26 Apr 2022 22:17:00 +0200
+Subject: [PATCH] TST: fix tests for GEOS main (#1357) (backported for nixpkgs)
+
+---
+ pygeos/tests/test_constructive.py | 6 +++++-
+ 1 file changed, 5 insertions(+), 1 deletion(-)
+
+diff --git a/pygeos/tests/test_constructive.py b/pygeos/tests/test_constructive.py
+index 87c0a9f..b3459d1 100644
+--- a/pygeos/tests/test_constructive.py
++++ b/pygeos/tests/test_constructive.py
+@@ -48,7 +48,11 @@ def test_no_args_array(geometry, func):
+ @pytest.mark.parametrize("geometry", all_types)
+ @pytest.mark.parametrize("func", CONSTRUCTIVE_FLOAT_ARG)
+ def test_float_arg_array(geometry, func):
+-    if func is pygeos.offset_curve and pygeos.get_type_id(geometry) not in [1, 2]:
++    if (
++        func is pygeos.offset_curve
++        and pygeos.get_type_id(geometry) not in [1, 2]
++        and pygeos.geos_version < (3, 11, 0)
++    ):
+         with pytest.raises(GEOSException, match="only accept linestrings"):
+             func([geometry, geometry], 0.0)
+         return
+-- 
+2.36.1
+

--- a/pkgs/development/python-modules/shapely/default.nix
+++ b/pkgs/development/python-modules/shapely/default.nix
@@ -2,6 +2,7 @@
 , stdenv
 , buildPythonPackage
 , fetchPypi
+, fetchpatch
 , substituteAll
 , pythonOlder
 , geos
@@ -42,6 +43,12 @@ buildPythonPackage rec {
       src = ./library-paths.patch;
       libgeos_c = GEOS_LIBRARY_PATH;
       libc = lib.optionalString (!stdenv.isDarwin) "${stdenv.cc.libc}/lib/libc${stdenv.hostPlatform.extensions.sharedLibrary}.6";
+    })
+    (fetchpatch {
+      name = "fix-tests-geos-3.11.patch";
+      url = "https://github.com/shapely/shapely/commit/21c8e8a7909e7fb3cce6daa5c5b8284ac927fcb0.patch";
+      includes = [ "tests/test_parallel_offset.py" ];
+      sha256 = "sha256-85c8NlmAzzfCgepe/411ug5Sq+665dFMb3ySaUt9Kew=";
     })
  ];
 

